test/gtest/CMakeLists.txt

```diff
@@ -1,23 +1,39 @@
 enable_testing()
 
+set(TEST_SOURCES)
+
 if (NOT ENABLE_ROCSPARSE)
-  add_executable(
-    spblas-tests
-    spmv_test.cpp
-    spmm_test.cpp
-    spgemm_test.cpp
-    spgemm_csr_csc.cpp
-    add_test.cpp
-    transpose_test.cpp
-    triangular_solve_test.cpp
-  )
-elseif(ENABLE_ROCSPARSE)
-  set_source_files_properties(rocsparse/spmv_test.cpp PROPERTIES LANGUAGE HIP)
-  add_executable(spblas-tests
-                 rocsparse/spmv_test.cpp)
+  list(APPEND TEST_SOURCES
+       spmv_test.cpp
+       spmm_test.cpp
+       spgemm_test.cpp
+       spgemm_csr_csc.cpp
+       add_test.cpp
+       transpose_test.cpp
+       triangular_solve_test.cpp)
 endif()
 
+if (SPBLAS_GPU_BACKEND)
+  list(APPEND TEST_SOURCES
+       thrust/spmv_test.cpp)
+
+  if (ENABLE_ROCSPARSE)
+    set_source_files_properties(${TEST_SOURCES} PROPERTIES LANGUAGE HIP)
+  endif()
+endif()
+
+add_executable(spblas-tests ${TEST_SOURCES})
+
 target_link_libraries(spblas-tests spblas fmt GTest::gtest_main)
 
+if (ENABLE_ROCSPARSE)
+  find_package(rocthrust REQUIRED)
```

Comment:
we had some challenges in previous experience on repeatedly finding the same package in different folders. I will suggest we just find the package once in the root CMakeLists and use it in the subfolders unless there is another concern.